AirCasting/CreateSessionViews/ChooseSessionTypeView.swift

```diff
@@ -119,18 +119,18 @@ struct ChooseSessionTypeView: View {
         }
         .background(
             Group {
-                NavigationLink(
-                    destination: PowerABView(),
-                    isActive: $isPowerABLinkActive,
-                    label: {
-                        EmptyView()
-                    })
-                NavigationLink(
-                    destination: TurnOnBluetoothView(),
-                    isActive: $isTurnBluetoothOnLinkActive,
-                    label: {
-                        EmptyView()
-                    })
+                EmptyView()
+                    .fullScreenCover(isPresented: $isPowerABLinkActive) {
+                        CreatingSessionFlowRootView {
+                            PowerABView()
+                        }
+                    }
+                EmptyView()
+                    .fullScreenCover(isPresented: $isTurnBluetoothOnLinkActive) {
+                        CreatingSessionFlowRootView {
```

Comment:
Could you explain what is happening here?

Reply:
Sure. It's complicated to explain as it was created because of some bugs I encountered.
The main bug was - when I pushed view as fullScreenCover I couldn't dismiss it. So I decided to add the cancel button. But, there's PowerAB view that can be the first fullScreenCover view (so it needs cancel button) or the second view, after TurnBluetoothOn view, and then it had Back button (it's in navigation view so it's added automatically). Adding the cancel button on this view resulted in having two buttons. There were other problems with that but these are the main two. 
So I created CreatingSessionFlowRootView that has a content of type View.  `content` is a closure that returns `Content` of type `View`. It has no initializers so when I initialize `CreatingSessionFlowRootView` I'm providing one of our views there. 
I added the cancel button in `CreatingSessionFlowRootView`. So now, instead of creating cancel buttons on TurnBluetoothOn and PowerAB and trying to find out when should I display cancel on PowerAB view, I'm creating it on CreatingSessionFlowRootView that displays the content (depending on the situation it's one of our views) and adds cancel button to the navigation bar.

Reply:
I get it. Thanks for the explanation :)